### ISSUE TITLE
Logical errors in Windows Server 2022 SCA checks

### ISSUE THREAD

**Issue by IsExec** [wazuh/wazuh#22536]
Some logical errors were detected in Windows Server 2022 SCA rules that could affect the correctness of the checks.
A total of 42 rules were identified to be affected by this logical error. The rules relates to the following sections:

S/N | Section
-- | --
1 | 2.3.2.1
2 | 2.3.6.1
3 | 2.3.6.2
4 | 2.3.6.3
5 | 2.3.6.6
6 | 2.3.7.7
7 | 2.3.8.2
8 | 2.3.9.1
9 | 2.3.9.4
10 | 2.3.10.2
11 | 2.3.10.10
12 | 2.3.10.11
13 | 2.3.11.5
14 | 2.3.11.8
15 | 2.3.15.1
16 | 2.3.15.2
17 | 2.3.17.5
18 | 2.3.17.6
19 | 2.3.17.7
20 | 2.3.17.8
21 | 9.1.1
22 | 9.1.2
23 | 9.2.1
24 | 9.2.2
25 | 9.3.1
26 | 9.3.2
27 | 18.4.2
28 | 18.4.3
29 | 18.5.6
30 | 18.5.8
31 | 18.5.9
32 | 18.5.12
33 | 18.4.6.2
34 | 18.7.8
35 | 18.9.35.2
36 | 18.9.38.1
37 | 18.10.57.3.9.4
38 | 18.10.57.3.9.5
39 | 18.10.57.3.11.1
40 | 18.10.57.3.11.2
41 | 18.10.87.1
42 | 18.10.93.4.1

**Comment by IsExec** [comment]
## Review Update - 18/03/2024

S/N | Section | Status
-- | -- | --
1 | 2.3.2.1 | Fixed
2 | 2.3.6.1 | Fixed
3 | 2.3.6.2 | Fixed
4 | 2.3.6.3 | Fixed
5 | 2.3.6.6 | Fixed
6 | 2.3.7.7 | Fixed
7 | 2.3.8.2 | Fixed

**Comment by Johnng007** [comment]
**22-03-2024**
7 of 42    logical errors fix confirmed
**25-03-2024**
20 of 42  logical errors fix confirmed
**26-03-2024**
34 0f 42  logical errors fix confirmed
**27-03-2024**
42 of 42  logical errors fix confirmed

**Comment by IsExec** [comment]
# Windows Server 2022 Rework SCA Testing Report
**Condition**: The new SCA was benchmarked against old Windows server 2022 SCA.  

## Agent running on old SCA (Windows server 2022 SCA)

###  Endpoint spec

<details>
<summary>CPU/RAM</summary>

````console
>systeminfo

Host Name:                 AGENTA
OS Name:                   Microsoft Windows Server 2022 Datacenter Evaluation
OS Version:                10.0.20348 N/A Build 20348
OS Manufacturer:           Microsoft Corporation
OS Configuration:          Standalone Server
OS Build Type:             Multiprocessor Free
Registered Owner:
Registered Organization:   Vagrant
Product ID:                00455-50000-00001-AA573
Original Install Date:     7/25/2024, 3:54:25 PM
System Boot Time:          7/25/2024, 6:41:05 PM
System Manufacturer:       innotek GmbH
System Model:              VirtualBox
System Type:               x64-based PC
Processor(s):              1 Processor(s) Installed.
                           [01]: Intel64 Family 6 Model 142 Stepping 10 GenuineIntel ~2112 Mhz
BIOS Version:              innotek GmbH VirtualBox, 12/1/2006
Windows Directory:         C:\Windows
System Directory:          C:\Windows\system32
Boot Device:               \Device\HarddiskVolume1
System Locale:             en-us;English (United States)
Input Locale:              en-us;English (United States)
Time Zone:                 (UTC-08:00) Pacific Time (US & Canada)
Total Physical Memory:     8,024 MB
Available Physical Memory: 4,961 MB
Virtual Memory: Max Size:  9,944 MB
Virtual Memory: Available: 7,262 MB
Virtual Memory: In Use:    2,682 MB
````

</details>

<details>
<summary>Disk</summary>

````console
Microsoft DiskPart version 10.0.20348.1

Copyright (C) Microsoft Corporation.
On computer: AGENTA

DISKPART> list volume

  Volume ###  Ltr  Label        Fs     Type        Size     Status     Info
  ----------  ---  -----------  -----  ----------  -------  ---------  --------
  Volume 0         boot         NTFS   Partition    350 MB  Healthy    System
  Volume 1     C   Windows 202  NTFS   Partition     59 GB  Healthy    Boot

>wmic logicaldisk get size,freespace,caption
Caption  FreeSpace    Size
C:       48257286144  64055406592
````
</details>


### Footprint test results
<details>
<summary>Metrics</summary>

![cpu_percent_plot](https://github.com/user-attachments/assets/ca4da185-47c0-4ffe-b2c5-91484ee93bec)
![disk_percent_plot](https://github.com/user-attachments/assets/76760eab-8dac-4530-bb85-29cdc36e0734)
![disk_used_(mb)_plot](https://github.com/user-attachments/assets/bc38007b-58fa-42ed-b1ce-3f06de56c4b5)
![memory_percent_plot](https://github.com/user-attachments/assets/5b79a865-c4cb-4b1e-be2a-1b11a2467a67)
![memory_used_(mb)_plot](https://github.com/user-attachments/assets/3a9f4274-954c-462e-94fc-794aa8da9cea)
![ProcessesCPU](https://github.com/user-attachments/assets/e4865b06-4a62-4a88-bd7f-e1bb169a70b2)
![ProcessesCPU_Change_Point](https://github.com/user-attachments/assets/728bcafc-dc3f-4d13-8058-007f3777a7ba)
![ProcessesCPUlegend](https://github.com/user-attachments/assets/5241236f-b6e9-4329-9403-8763a7bc6f5e)
![ProcessesRAM](https://github.com/user-attachments/assets/75fa165a-9442-4d03-a16a-9501f864eaad)
![ProcessesRAM_Change_Point](https://github.com/user-attachments/assets/a17213ea-d205-4c8f-84f7-d1294fee20b1)
![ProcessesRAMlegend](https://github.com/user-attachments/assets/ad1db3d6-bcca-4567-8b33-292613f1f89b)
![swap_percent_plot](https://github.com/user-attachments/assets/fb3d44e2-c9b7-4a30-8579-0d0e231ac91b)
![swap_used_(mb)_plot](https://github.com/user-attachments/assets/6ca46b2b-29ff-4af2-964a-eee249e51a19)
</details>

<details>
<summary>SCA logs</summary>

````console
2024/07/26 02:25:52 sca: INFO: Starting evaluation of policy: 'C:\Program Files (x86)\ossec-agent\ruleset\sca\cis_win2022.yml'
2024/07/26 02:26:03 sca: INFO: Evaluation finished for policy 'C:\Program Files (x86)\ossec-agent\ruleset\sca\cis_win2022.yml'
2024/07/26 02:26:03 sca: INFO: Security Configuration Assessment scan finished. Duration: 11 seconds.
2024/07/26 02:30:52 sca: INFO: Starting Security Configuration Assessment scan.
2024/07/26 02:30:52 sca: INFO: Starting evaluation of policy: 'C:\Program Files (x86)\ossec-agent\ruleset\sca\cis_win2022.yml'
2024/07/26 02:31:01 sca: INFO: Evaluation finished for policy 'C:\Program Files (x86)\ossec-agent\ruleset\sca\cis_win2022.yml'
2024/07/26 02:31:01 sca: INFO: Security Configuration Assessment scan finished. Duration: 9 seconds.
2024/07/26 02:35:52 sca: INFO: Starting Security Configuration Assessment scan.
2024/07/26 02:35:52 sca: INFO: Starting evaluation of policy: 'C:\Program Files (x86)\ossec-agent\ruleset\sca\cis_win2022.yml'
2024/07/26 02:36:00 sca: INFO: Evaluation finished for policy 'C:\Program Files (x86)\ossec-agent\ruleset\sca\cis_win2022.yml'
2024/07/26 02:36:00 sca: INFO: Security Configuration Assessment scan finished. Duration: 8 seconds.
````

</details>

[metrics.zip](https://github.com/user-attachments/files/16390114/metrics.zip)


## Agent running on new SCA (Windows server 2022 SCA Rework)

### Endpoint spec
<details>
<summary>CPU/RAM</summary>

````console
>systeminfo

Host Name:                 AGENTA
OS Name:                   Microsoft Windows Server 2022 Datacenter Evaluation
OS Version:                10.0.20348 N/A Build 20348
OS Manufacturer:           Microsoft Corporation
OS Configuration:          Standalone Server
OS Build Type:             Multiprocessor Free
Registered Owner:
Registered Organization:   Vagrant
Product ID:                00455-50000-00001-AA573
Original Install Date:     7/25/2024, 3:54:25 PM
System Boot Time:          7/25/2024, 6:41:05 PM
System Manufacturer:       innotek GmbH
System Model:              VirtualBox
System Type:               x64-based PC
Processor(s):              1 Processor(s) Installed.
                           [01]: Intel64 Family 6 Model 142 Stepping 10 GenuineIntel ~2112 Mhz
BIOS Version:              innotek GmbH VirtualBox, 12/1/2006
Windows Directory:         C:\Windows
System Directory:          C:\Windows\system32
Boot Device:               \Device\HarddiskVolume1
System Locale:             en-us;English (United States)
Input Locale:              en-us;English (United States)
Time Zone:                 (UTC-08:00) Pacific Time (US & Canada)
Total Physical Memory:     8,024 MB
Available Physical Memory: 4,961 MB
Virtual Memory: Max Size:  9,944 MB
Virtual Memory: Available: 7,262 MB
Virtual Memory: In Use:    2,682 MB
````

</details>

<details>
<summary>Disk</summary>

````console
Microsoft DiskPart version 10.0.20348.1

Copyright (C) Microsoft Corporation.
On computer: AGENTA

DISKPART> list volume

  Volume ###  Ltr  Label        Fs     Type        Size     Status     Info
  ----------  ---  -----------  -----  ----------  -------  ---------  --------
  Volume 0         boot         NTFS   Partition    350 MB  Healthy    System
  Volume 1     C   Windows 202  NTFS   Partition     59 GB  Healthy    Boot

>wmic logicaldisk get size,freespace,caption
Caption  FreeSpace    Size
C:       48257286144  64055406592

````
</details>


### Footprint test results
<details>
<summary>Metrics</summary>

![cpu_percent_plot](https://github.com/user-attachments/assets/0c723a31-728f-4e57-98f2-e52623792d70)
![disk_percent_plot](https://github.com/user-attachments/assets/94b16e47-ef8f-4b66-bc95-35e362bd053d)
![disk_used_(mb)_plot](https://github.com/user-attachments/assets/b254b7e8-ed38-45b1-aa82-48a14feedce2)
![memory_percent_plot](https://github.com/user-attachments/assets/c6969c96-3ea2-480e-afd7-c85110badd9a)
![memory_used_(mb)_plot](https://github.com/user-attachments/assets/a9f2fb68-e7d5-477e-8f08-b96ef9cada5f)
![ProcessesCPU](https://github.com/user-attachments/assets/d8a75c5f-4e35-48da-91c8-e92f960cafb1)
![ProcessesCPU_Change_Point](https://github.com/user-attachments/assets/ca3b724a-a0bb-4b93-8011-a26b9ef69992)
![ProcessesCPUlegend](https://github.com/user-attachments/assets/0470bf65-e39a-4531-bd98-b03b15ceba1a)
![ProcessesRAM](https://github.com/user-attachments/assets/dec6cf08-1439-4cb3-83e0-60fb619c70c8)
![ProcessesRAM_Change_Point](https://github.com/user-attachments/assets/7544170e-96ef-4987-8386-71e04735c6d8)
![ProcessesRAMlegend](https://github.com/user-attachments/assets/cc117673-1b9d-452e-94db-ad144bdca210)
![swap_percent_plot](https://github.com/user-attachments/assets/3bd94350-825f-4410-8406-0664d5e8c034)
![swap_used_(mb)_plot](https://github.com/user-attachments/assets/9d7f7ab0-8999-4910-a478-e4f66f377ba5)
</details>

<details>
<summary>SCA logs</summary>

````console
2024/07/26 01:39:49 sca: INFO: Starting Security Configuration Assessment scan.
2024/07/26 01:39:49 sca: INFO: Starting evaluation of policy: 'C:\Program Files (x86)\ossec-agent\ruleset\sca\cis_win2022.yml'
2024/07/26 01:39:59 sca: INFO: Evaluation finished for policy 'C:\Program Files (x86)\ossec-agent\ruleset\sca\cis_win2022.yml'
2024/07/26 01:39:59 sca: INFO: Security Configuration Assessment scan finished. Duration: 10 seconds.
2024/07/26 01:44:49 sca: INFO: Starting Security Configuration Assessment scan.
2024/07/26 01:44:49 sca: INFO: Starting evaluation of policy: 'C:\Program Files (x86)\ossec-agent\ruleset\sca\cis_win2022.yml'
2024/07/26 01:45:00 sca: INFO: Evaluation finished for policy 'C:\Program Files (x86)\ossec-agent\ruleset\sca\cis_win2022.yml'
2024/07/26 01:45:00 sca: INFO: Security Configuration Assessment scan finished. Duration: 11 seconds.
2024/07/26 01:49:50 sca: INFO: Starting Security Configuration Assessment scan.
2024/07/26 01:49:50 sca: INFO: Starting evaluation of policy: 'C:\Program Files (x86)\ossec-agent\ruleset\sca\cis_win2022.yml'
2024/07/26 01:50:17 sca: INFO: Evaluation finished for policy 'C:\Program Files (x86)\ossec-agent\ruleset\sca\cis_win2022.yml'
2024/07/26 01:50:17 sca: INFO: Security Configuration Assessment scan finished. Duration: 27 seconds.

````

</details>

[metrics.zip](https://github.com/user-attachments/files/16390173/metrics.zip)


## Analysis

  | Benchmark (Win   server 2022 old SCA) | New SCA (Win   server 2022 rework) | Difference
-- | -- | -- | --
CPU |   |   |  
wazuh-agent.exe average   CPU percent | 1.4 | 1.4 | 0
Overall   average CPU percent | 20.73 | 19.27 | -1.46
Overall peak   CPU percent | 49.5 | 39.8 | -9.7
Memory |   |   |  
wazuh-agent.exe   average memory (MB) | 0 | 0 | 0
Overall   average memory used (MB) | 1454.43 | 1434.12 | -20.31
Overall peak   memory used peak (MB) | 1457.89 | 1439.14 | -18.75
Disk |   |   |  
Overall   average disk used (MB) | 19391.09 | 19233.92 | -157.17
Overall peak   disk used (MB) | 19391.09 | 19233.92 | -157.17
Overall   average disk read bytes | 4468.3 | 4257.29 | -211.01
Overall peak   disk read bytes | 4468.3 | 4257.29 | -211.01
Overall   average disk write bytes | 2830.15 | 1361.59 | -1468.56
Overall peak   disk write bytes | 2830.28 | 1361.71 | -1468.57

**Comment by ooniagbi** [comment]
Merged to 4.10.0
